examples/src/main/java/io/grpc/examples/routeguide/RouteGuideClient.java

```diff
@@ -265,18 +288,51 @@ public static void main(String[] args) throws InterruptedException {
       client.recordRoute(features, 10);
 
       // Send and receive some notes.
-      client.routeChat();
+      client.routeChat().await(1, TimeUnit.MINUTES);
     } finally {
       client.shutdown();
     }
   }
 
-  private static void info(String msg, Object... params) {
+  private void info(String msg, Object... params) {
     logger.log(Level.INFO, msg, params);
   }
 
+  private void warning(String msg, Object... params) {
+    logger.log(Level.WARNING, msg, params);
+  }
+
+  /**
+   * Only used for unit test, as we do not want to introduce randomness in unit test.
+   */
+  @VisibleForTesting
+  void setRandom(Random random) {
+    this.random = random;
+  }
+
   private RouteNote newNote(String message, int lat, int lon) {
     return RouteNote.newBuilder().setMessage(message)
         .setLocation(Point.newBuilder().setLatitude(lat).setLongitude(lon).build()).build();
   }
+
+  /**
+   * Only used for helping unit test.
+   */
+  @VisibleForTesting
+  interface TestHelper {
```

Comment:
That might be a little confusing or distracting, `StreamObserver` has its own use but test helping. After all, it's adding only one interface.